mono_repo/lib/src/commands/github/github_yaml.dart

```diff
@@ -43,14 +43,38 @@ Map<String, String> generateGitHubYml(
       );
     }
 
-    myJobs = myJobs.toList()
-      ..sort((a, b) => orderedStages
-          .indexOf(a.stageName)
-          .compareTo(orderedStages.indexOf(b.stageName)));
+    final sortedJobs = myJobs.toList()
+      ..sort((a, b) {
+        var value = orderedStages
+            .indexOf(a.stageName)
+            .compareTo(orderedStages.indexOf(b.stageName));
+
+        if (value == 0) {
+          if (a is _SelfValidateJob) {
+            value = -1;
+          }
+          if (b is _SelfValidateJob) {
+            value = 1;
+          }
+        }
+
+        if (value == 0 && a is CIJob && b is CIJob) {
+          value = a.sortBits.compareTo(b.sortBits);
+        }
+        if (value == 0) {
+          assert(() {
+            print(
```

Comment:
Why hide this behind an assert? I don't think anybody really runs with asserts enabled... and this won't fail either so the one situation (tests) where it would trigger by default we just won't see it?

Reply:
Ack. I could just assert.

Otherwise, LGTY @jakemac53 ?